packages/web/src/pages/search-page-v2/BpmFilter.tsx

```diff
@@ -82,9 +84,18 @@ type ViewProps = {
   handleChange: (value: string, label: string) => void
 }
 
+function useStateDebounced<T>(initialValue: T) {
+  const [value, setValue] = useState(initialValue)
+  const setValueDebounced = useMemo(
+    () => debounce(setValue, DEBOUNCE_TIME),
+    [setValue]
+  )
+  return [value, setValueDebounced] as const
+}
```

Comment:
oh im surprised that we don't already have a hook for this. do we want to add this to the hooks folder?

Reply:
We have `useDebouncedCallback` but that's slightly different. Yeah I'll add it!